src/o365/spo/commands/listitem/listitem-record-declare.ts

```diff
@@ -0,0 +1,309 @@
+              `Retrieved access token ${accessToken}. Retrieving request digest...`
+            );
+          }
+
+          return this.getRequestDigest(cmd, this.debug);
+        })
+      .then(
+        (contextResponse: ContextInfo): Promise<IdentityResponse> => {
+          formDigestValue = contextResponse.FormDigestValue;
+
+          if (this.debug) {
+            cmd.log("contextResponse:");
+            cmd.log(JSON.stringify(contextResponse));
+            cmd.log("");
+          }
+
+          return clientSvc.getCurrentWebIdentity(
+            args.options.webUrl,
+            siteAccessToken,
+            formDigestValue
+          );
+        }
+      )
+      .then((webIdentityResp: IdentityResponse): request.RequestPromise => {
+        webIdentity = webIdentityResp.objectIdentity;
+
+        const requestOptions: any = {
+          url: `${listRestUrl}?$select=Id`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            accept: 'application/json;odata=nometadata'
+          }),
+          json: true
+        }
+
+        if (this.debug) {
+          cmd.log('Executing get list web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((res: { Id: string }): request.RequestPromise => {
+
+        listId = res.Id;
+        const requestBody = this.generateDeclareRecordRequestBody(webIdentity, listId, id, dateArgument);
+
+        const requestOptions: any = {
+          url: `${args.options.webUrl}/_vti_bin/client.svc/ProcessQuery`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${siteAccessToken}`,
+            'Content-Type': 'text/xml',
+            'X-RequestDigest': formDigestValue
+          }),
+          body: requestBody
+        };
+
+        if (this.debug) {
+          cmd.log("Executing declare item as record web request...");
+          cmd.log(requestOptions);
+          cmd.log("");
+          cmd.log("Body:");
+          cmd.log(requestBody);
+          cmd.log("");
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: string): void => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        const json: ClientSvcResponse = JSON.parse(res);
+        const response: ClientSvcResponseContents = json[0];
+
+        if (response.ErrorInfo) {
+          cmd.log(new CommandError(response.ErrorInfo.ErrorMessage));
+        }
+        else {
+          const result: boolean = json[json.length - 1];
+          cmd.log(result);
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
+  }
+
+  protected generateDeclareRecordRequestBody(webIdentity: string, listId: string, id: string, date: string): string {
+
+    let requestBody = "";
+    if (date.length == 10) {
```

Comment:
We should use type-safe comparison, eg. `===` instead of `==`

Reply:
Done!